### PR TITLE
[ENH] Uploads cleanup and issues

### PR DESCRIPTION
**Pull request by rksh**
- templating values in presentation layer instead of wonky string concat
- making layouts w/ direction for presentation work right in... more cases.
- adding _iw strings as copy of _he to accommodate old Sun weirdness in i18n
- platting status on main list view into DB and Scan cells, templating values
- adding "x scanned in y" to dash
- some incidental language file improvements from testing
- using padding in layouts instead of wonky manual spacing

### CODE REVIEW

File: wiglewifiwardriving/src/main/res/values-zh-rHK/strings.xml [hunk 1]
```diff
@@ -119,19 +119,19 @@
...
     <string name="dash_dist_run">運行距離：</string>
     <string name="dash_time_run">運行時間：</string>
     <string name="dash_time_scanning">掃描持續時間：</string>
     <string name="dash_dist_total">共有區。</string>
     <string name="dash_dist_prev">以前的運行：</string>
-    <string name="dash_db_queue">DB的隊列：</string>
-    <string name="dash_db_nets">DB的籃網：</string>
-    <string name="dash_db_locs">DB的位置：</string>
-    <string name="dash_short_loc">LOC：</string>
+    <string name="dash_db_queue">DB的隊列: %1$s</string>
```
Comment (uhtu) [inline wiglewifiwardriving/src/main/res/values-zh-rHK/strings.xml:131]: these : appear to be different characters? (this one vs the three below are not the same)
(this one is colon, the other three are FULLWIDTH COLON 😨 )

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java [hunk 2]
```diff
@@ -481,10 +504,10 @@ private void handleScanChange(final MainActivity main, final View view ) {
         final boolean isScanning = main == null || main.isScanning();
         Logging.info("list handleScanChange: isScanning now: " + isScanning );
         if ( isScanning ) {
-            setStatusUI(view, getString(R.string.list_scanning_on));
+        setScanStatusUI(view, getString(R.string.list_scanning_on));
```
Comment (bobzilladev) [inline wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java:507]: missing indent omg!

**Review by bobzilladev — APPROVED** [review]
In addition to uhtu's actually useful feedback, I have one vital comment that will make or break this entire app!

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DashboardFragment.java [hunk 3]
```diff
@@ -138,50 +136,56 @@ private void updateUI( final View view ) {
             TextView dashScanstatus = view.findViewById(R.id.dash_scanstatus);
             dashScanstatus.setText(getString(R.string.dash_scan_off));
         }
+        final NumberFormat n = NumberFormat.getInstance();
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/DashboardFragment.java:139]: that one's decimal, but I can move this one up.

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java [hunk 4]
```diff
@@ -1783,8 +1785,10 @@ public void scheduleScan() {
     }
 
     public void speak(final String string) {
-        if (!MainActivity.getMainActivity().isMuted() && state.tts != null) {
-            state.tts.speak(string, TextToSpeech.QUEUE_ADD, null);
+        final MainActivity a = MainActivity.getMainActivity();
+        if (a != null && !a.isMuted() && state.tts != null) {
+            state.tts.speak(string, TextToSpeech.QUEUE_ADD, null,
+                    "WiGLEtts-"+utteranceSequenceGenerator.getAndAdd(1L));
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/MainActivity.java:1791]: old method is deprecated - you have to put something there, and it sounds like things go badly if it's not unique-ish.

File: wiglewifiwardriving/src/main/res/layout/dashtop.xml [hunk 5]
```diff
@@ -247,20 +253,22 @@
         <TextView
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
-            android:paddingLeft="3dp"
+            android:paddingStart="3dp"
             android:text="@string/dash_time_scanning"
-            android:textSize="30dp"
+            android:textSize="30sp"
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/res/layout/dashtop.xml:258]: sp for text, dp for dots.

File: wiglewifiwardriving/src/main/res/values-ar/strings.xml [hunk 6]
```diff
@@ -119,34 +119,34 @@
     <string name="cancel">إلغاء</string>
     <string name="dash_scan_off">(تم إيقاف تشغيل المسح)</string>
     <string name="dash_new_wifi">واي فاي جديد</string>
-    <string name="dash_vis_nets">شبكات مرئية:</string>
-    <string name="dash_new_upload">شبكات جديدة منذ التحميل:</string>
+    <string name="dash_vis_nets">شبكات مرئية: %1$d</string>
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/res/values-ar/strings.xml:122]: these are adjacent in a horrible way, thanks unicode.

File: wiglewifiwardriving/src/main/res/values-he/strings.xml [hunk 7]
```diff
@@ -119,19 +119,19 @@
     <string name="cancel">לבטל</string>
     <string name="dash_scan_off">(סריקה מבוטלת)</string>
     <string name="dash_new_wifi">חדש WiFi</string>
-    <string name="dash_vis_nets">Visible נטס:</string>
-    <string name="dash_new_upload">הנטס בניו מאז טען:</string>
+    <string name="dash_vis_nets">%1$d Visible נטס: </string>
+    <string name="dash_new_upload">הנטס בניו מאז טען: %1$d</string>
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/res/values-he/strings.xml:123]: also adjacent in a horrible way.

File: wiglewifiwardriving/src/main/res/values-iw/strings.xml [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+    <string name="dash_new_bt">Bluetooth חדש</string>
+    <string name="dash_dist_run">הפעל מרחק:</string>
+    <string name="dash_time_run">משך הפעלה:</string>
+    <string name="dash_time_scanning">משך הסריקה:</string>
+    <string name="dash_dist_total">סך Dist.:</string>
+    <string name="dash_dist_prev">הקודם הפעלה:</string>
+    <string name="dash_db_queue">%1$s :DB תור</string>
+    <string name="dash_db_nets">%1$s :DB נטס</string>
+    <string name="dash_db_locs">%1$s :DB מיקומים</string>
+    <string name="dash_short_loc">לוק: %1$s</string>
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/res/values-iw/strings.xml:134]: also adjacent in a horrible way.

File: wiglewifiwardriving/src/main/res/values-iw/strings.xml [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+    <string name="firsttransid_title">מזהה עסקה ראשון</string>
+    <string name="lasttransid_title">מזהה עסקה אחרון</string>
+    <string name="gps_permission">GPS (מיקום יפה)</string>
+    <string name="cell_permission">מיקום סלולרי (מיקום גס)</string>
+    <string name="rank_stats_app_name">WiGLE.net דירוגים</string>
+    <string name="total_cell">סה"כ תא</string>
+    <string name="tab_rank">דירוגים</string>
+    <string name="uploads_app_name">העלאות של WiGLE.net</string>
+    <string name="tab_uploads">העלאות</string>
+    <string name="queue_depth">תור WiGLE: %1$s Tri: %2$s גיאוגרפי: %3$s</string>
```
[the remark this answers is not in the document]
Comment (rksh) [reply wiglewifiwardriving/src/main/res/values-iw/strings.xml:298]: also adjacent in a horrible way.

File: wiglewifiwardriving/src/main/res/values-zh-rHK/strings.xml (hunk 1 above)
Comment (rksh) [reply wiglewifiwardriving/src/main/res/values-zh-rHK/strings.xml:131]: solid catch

File: wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java (hunk 2 above)
Comment (uhtu) [reply wiglewifiwardriving/src/main/java/net/wigle/wigleandroid/ListFragment.java]: :rage4:

**Review by uhtu — APPROVED** [review]

**Pull request merged**